source/data_at_rest_encryption.rst

```diff
@@ -150,6 +132,54 @@ All these options can be specified in the configuration file:
    |mongodb| Documentation: How to set options in a configuration file
       https://docs.mongodb.com/manual/reference/configuration-options/index.html#configuration-file
 
+.. _psmdb.data-at-rest-encryption-vault:
+
+HashiCorp Vault Integration
+================================================================================
+
+Starting from version 3.6.13-3.3, |PSMDB| provides |vault| integration. We only
+support the |vault| backend with KV Secrets Engine - Version 2 (API)
+with versioning enabled.
+
+.. seealso::
+
+   How to configure the KV Engine
+      https://www.vaultproject.io/api/secret/kv/kv-v2.html
+
+==========================  ====================================  ==========
+Command Line	            Config File                           Type
+==========================  ====================================  ==========
+vaultServerName	            security.vault.ServerName	          string
+vaultPort	            security.vault.port	                  int
+vaultTokenFile	            security.vault.secret	          string
+vaultSecret	            security.vault.secret	          string
+vaultRotateMasterKey	    security.vault.vaultrotateMasterKey	  switch
+vaultServerCAFile	    security.vault.serverCAFile	          string
+vaultDisableTLSForTesting   security.vault.disableTLSForTesting	  switch
+==========================  ====================================  ==========
+
+The vault token file consists of the raw vault token and does not include any
+additional strings or parameters.
+
+On start, the server tries to read the master key from the Vault. If the
+configured secret does not exist, Vault responds with the HTTP 404 error. During
+the first run of |PSMDB|, the process generates a secure key and writes the key
+to the vault.
+
+Migrating from Key File Encryption to |vault| Encryption
+--------------------------------------------------------------------------------
+
+You can use either the |vault| integration feature or the key file
+encryption. If you have used the key file encryption and wish to migrate to  
+using |vault|, complete the following steps:
+
+1. Insert the key from keyfile into the |vault| server to the desired secret
+   path.
+#. Stop mongod.
+#. Add the startup options for |vault| integration for data at rest encryption.
+#. Remove the startup options for the key file encryption.
+#. Start the ``mongod`` service
```

Comment:
Well, this procedure should work but I think it would be great to provide some examples along the way. How to insert key into the vault? Which parameters should be added for vault and removed for keyfile?